File: src/main/java/org/dxworks/utils/java/rest/client/providers/CookieAuthenticationProvider.java
package org.dxworks.utils.java.rest.client.providers;

import com.google.api.client.http.HttpRequest;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class CookieAuthenticationProvider implements AuthenticationProvider {
    private String cookie;

    @Override
    public void initialize(HttpRequest httpRequest) {
        httpRequest.getHeaders().setCookie(cookie);
    }
}

File: src/main/kotlin/org/dxworks/utils/java/rest/client/providers/CookieAuthenticationProvider.kt
package org.dxworks.utils.java.rest.client.providers

import com.google.api.client.http.HttpRequest

open class CookieAuthenticationProvider(protected val cookie: String) : AuthenticationProvider {
    override fun initialize(httpRequest: HttpRequest) {
        httpRequest.headers.cookie = cookie
    }
}
